### PR TITLE
add serverless deployment template

### PR DESCRIPTION
**Pull request by daidoli**
1. remove unused graphql package
2. await `makerRoutine`
3. add serverless.yaml
4. export `main`

### CODE REVIEW

File: .env.staging [hunk 1]
```diff
@@ -0,0 +1,7 @@
+STAGE=staging
```
[the remark this answers is not in the document]
Comment (daidoli) [reply .env.staging]: fixed

File: .env.staging [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+REGION=
```
[the remark this answers is not in the document]
Comment (daidoli) [reply .env.staging]: removed `REGION`

File: .env.staging [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+AWS_REGION=
+
+NETWORK=optimism-kovan
+L2_WEB3_ENDPOINT=
```
[the remark this answers is not in the document]
Comment (daidoli) [reply .env.staging]: fixed

File: README.md [hunk 4]
```diff
@@ -69,3 +69,25 @@ $ env $(cat .env | grep -v '#' | xargs) npm run start
...
 ```
+
+## Deployment
+
+### AWS Lambda
+
+Prerequisite
+
+-   `~/.aws/credentials` should have default profile with `aws_access_key_id` and `aws_secret_access_key`
+-   Fill in envs in `.env.staging`
```
[the remark this answers is not in the document]
Comment (daidoli) [reply README.md]: resolved

File: package.json [hunk 5]
```diff
@@ -5,7 +5,10 @@
     "main": "index.js",
     "scripts": {
         "start": "ts-node src/index.ts",
-        "build": "tsc -b ."
+        "build": "tsc -b .",
+        "sls:deploy:staging": "sls deploy --stage staging",
```
[the remark this answers is not in the document]
Comment (daidoli) [reply package.json]: resolved

File: serverless.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+plugins:
+    - serverless-plugin-log-retention
+    - serverless-webpack
+
+useDotenv: true
+
+provider:
+    name: aws
+    runtime: nodejs14.x
```
[the remark this answers is not in the document]
Comment (daidoli) [reply serverless.yml]: updated to nodejs 16 which is just released 2 weeks ago

File: .env.example [hunk 7]
```diff
@@ -3,3 +3,7 @@ L2_WEB3_ENDPOINTS=
 
 # wallet private key
 PRIVATE_KEY=
+
+STAGE=production
```
Comment (CheshireCatNick) [inline .env.example:7]: I think we can remove line 7 and 8 since we set it in index.ts. Just to reduce some confusion.

File: .gitignore [hunk 8]
```diff
@@ -71,6 +71,7 @@ typings/
 # dotenv environment variables file
 .env
 .env.test
+.env.staging
```
Comment (CheshireCatNick) [inline .gitignore:74]: can remove this

File: .env.example [hunk 9]
```diff
@@ -3,3 +3,7 @@ L2_WEB3_ENDPOINTS=
...
+NETWORK=optimism
+AWS_REGION=
```
Comment (CheshireCatNick) [inline .env.example:9]: can add a comment here to emphasize this is optional and only for deployment.

File: README.md [hunk 10]
```diff
@@ -69,3 +68,26 @@ $ env $(cat .env | grep -v '#' | xargs) npm run start
...
+
+-   `~/.aws/credentials` should have default profile with `aws_access_key_id` and `aws_secret_access_key`
+-   copy `.env.example` to `.env`
+-   Fill in envs in `.env`
+
+Deploy
+
+```bash
+npm run build
+npm run sls:deploy:staging
```
Comment (CheshireCatNick) [inline README.md:86]: should be sls:deploy

File: README.md [hunk 11]
```diff
@@ -69,3 +68,26 @@ $ env $(cat .env | grep -v '#' | xargs) npm run start
...
+```
+
+See CloudWatch log
+
+```bash
+AWS_REGION=${YOUR_REGION} npm run sls:log:staging
```
Comment (CheshireCatNick) [inline README.md:92]: should be sls:log

File: serverless.yml [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+provider:
+    name: aws
+    runtime: nodejs16.x
+    memorySize: 512
+    versionFunctions: false
+    stage: ${opt:stage, 'production'}
+    region: ${env:AWS_REGION}
+    environment:
+        STAGE: ${opt:stage, 'production'}
```
Comment (CheshireCatNick) [inline serverless.yml:17]: do we need STAGE and NETWORK env here? We already set it in index.ts

**Review by CheshireCatNick — CHANGES_REQUESTED** [review]

File: .env.example (hunk 7 above)
Comment (s60912frank) [reply .env.example]: fixed

File: .gitignore (hunk 8 above)
Comment (s60912frank) [reply .gitignore]: fixed

File: .env.example (hunk 9 above)
Comment (s60912frank) [reply .env.example:8]: fixed

File: README.md (hunk 10 above)
Comment (s60912frank) [reply README.md]: fixed

File: README.md (hunk 11 above)
Comment (s60912frank) [reply README.md]: fixed

**Review by CheshireCatNick — APPROVED** [review]

**Pull request merged**